ibis-server/app/model/metadata_dto.py

```diff
@@ -0,0 +1,114 @@
+from enum import Enum
+from app.model.data_source import (
+    PostgresConnectionUrl,
+    PostgresConnectionInfo,
+    BigQueryConnectionInfo,
+    SnowflakeConnectionInfo,
+)
+from pydantic import BaseModel, Field
+from typing import List, Optional, Dict, Any, Union
+
+
+class MetadataDTO(BaseModel):
+    connection_info: Union[
+        PostgresConnectionUrl | PostgresConnectionInfo,
+        BigQueryConnectionInfo,
+        SnowflakeConnectionInfo,
+    ] = Field(alias="connectionInfo")
```

Comment:
You could just use `connection_info: ConnectionInfo = Field(alias="connectionInfo")`